test/extensions/tracers/datadog/naming_test.cc

```diff
@@ -0,0 +1,223 @@
+/**
+ * The tests in this file aren't specific to a class, but instead test all
+ * behavior related to spans' "operation name" (a.k.a. "span name"),
```

Comment:
nit: if this is part of span, consider adding to the `span_test.cc` file (there are a few test files with multiple "Test" classes). This will allow quicker discovery of span relevant tests.

Reply:
This test involves both `Tracer` and `Span`. I could justify adding it to either of those drivers, but I prefer to leave it separate in order to emphasize the particular behavior under test.

`tracer_test.cc` and `span_test.cc` together cover the code paths under test here. I've added this new driver to ensure that the particular overall behavior of "operation name, "resource name," and "service name" is maintained.